### DISCUSSION TITLE
No readings from Lighthouse (LH Deck has LED always on Yellow)

### DISCUSSION THREAD

**Discussion by mouhyemen** [bitcraze/discussions#251] [Q&A] [answered]
I have been using CF2.1 and LH base stations V1 for the last 2+ years without fail.

I recently moved to a new test bed area where there are large mirrors in the room. At times, the state estimation would get off severely. This prompted me to switch the lighthouse deck with a newer one and/or re-calibrate the CF. After many failed attempts, I decided to flash the latest firmware onto the CF2.1.

**That brings me to the current unsolved issue now.**

I installed the latest cfclient (and cflib), I upgraded the CF2.1 to the latest firmware (2022.05), and my CF2.1 does not read any LH data.

![cf_lighthousedeck_yellow_led](https://user-images.githubusercontent.com/10107344/188221163-6d8769c9-fc06-4f86-9900-9c509069d6df.jpg)

The yellow LED is always on after turning on the CF2.1. Also, here's the output from the console tab.

```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:8c606d1f8f89 (2022.05) CLEAN
SYS: I am 0x20373357594B50070025002C and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcLighthouse4 for deck 0
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.0
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 19240 bytes
LHFL: Lighthouse bootloader version: 2
STAB: Ready to fly.
LHFL: Bitstream CRC32: 112BC794 [PASS]
LHFL: Firmware version 6 verified, booting deck!
LH: Warning: not getting data from all sensors
LH:   0 - error
LH:   1 - OK
LH:   2 - OK
LH:   3 - OK
```
How do i proceed?

**Comment by knmcguire** [comment]
Hi!

Some additional questions to your problem:
- The picture you showed, the red LED was on. Was it a solid on all the time?
- If you look at Sensor 0 on the lighthouse deck, can you see something off with any of the connections? Could you provide an close up of that sensor?
- Have you attempted to block off the mirrors? mirrors or reflections are actually quite bad for the lighthouse system

**Reply by mouhyemen** [reply, marked as answer]
- **The picture you showed, the red LED was on. Was it a solid on all the time?**
Are you referring to M1? No, it would blink.

- **If you look at Sensor 0 on the lighthouse deck, can you see something off with any of the connections? Could you provide an close up of that sensor?**
I don't think there's anything off with the connection. The same deck worked perfectly well in another test bed setting.

- **Have you attempted to block off the mirrors? mirrors or reflections are actually quite bad for the lighthouse system**
I have tried blocking off a glass wall. I need to work on that part further.

I was eventually able to get the LED out of yellow to green. 

I re-installed all the libraries, followed the steps in the tutorial, and then the LED turned from yellow to green. But that brought to another issue of Kalman prediction rate being low for which I opened a new discussion ([link](https://github.com/orgs/bitcraze/discussions/253)).

I think we can close this issue.

**Reply by knmcguire** [reply]
Ah good! 
Perhaps you can quote the 'I re-installed all the libraries..... opened a new discussion' line in a 'Suggest an answer' underneath here and mark that as an answer to close it properly

**Reply by David-Glom** [reply]
Hi have the same problem, but **not** the problem with sensor 0. But the light always stays yellow. Using another deck works, so there has to be some problem with the deck. Are there any fixes?

**Reply by knmcguire** [reply]
Hi @David-Glom 

Could you start a new discussion please, since this one has already been completed, and it seems you have a different issue.

 Also indicate if you have gone through the getting started tutorial of the lighthouse system: https://www.bitcraze.io/documentation/tutorials/getting-started-with-lighthouse/
